test/distributed/_tensor/test_dtensor.py

```diff
@@ -542,6 +546,51 @@ def test_dtensor_save_load(self):
         reloaded_st = torch.load(buffer, weights_only=True)
         self.assertEqual(sharded_tensor, reloaded_st)
 
+    @with_comms
+    @unittest.skipIf(
+        IS_FBCODE,
+        "subprocess import torch fails with ModuleNotFoundError: No module named 'torch' in fbcode",
+    )
+    def test_dtensor_save_load_import(self):
+        for should_import in [True, False]:
+            device_mesh = self.build_device_mesh()
+            placements = [Shard(0)]
+            local_tensor = torch.randn(3, 3)
+            sharded_tensor = DTensor.from_local(local_tensor, device_mesh, placements)
+            with tempfile.NamedTemporaryFile() as f:
+                torch.save(sharded_tensor, f)
+                import_string = (
+                    "import torch.distributed.tensor" if should_import else ""
+                )
+                script = f"""
+import torch
+{import_string}
+x = torch.load("{f.name}", weights_only=True)
+"""
+                try:
+                    subprocess.check_output(
+                        [sys.executable, "-c", script],
+                        cwd=os.path.dirname(os.path.realpath(__file__)),
+                        stderr=subprocess.STDOUT,
+                    )
+                    if not should_import:
+                        raise RuntimeError(
+                            "Script executed successfully despite lack of import"
+                        )
```

Comment:
How about something like that:
```python
cm = self.assertRaisesRegex(RuntimeError, "_picke.Unpickling...") else contextlib.nullcontext()
with cm:
    try:
      subprocess.check_call([sys.executable, "-c", script])
  except subprocess.CalledProcessError as e:
      raise RuntimeError(e.output.decode("utf-8"))
```
This way you can account for case when something should finish without an error and when something should raise an exception